### PR TITLE
#206: default parcel page sorting

### PR DESCRIPTION
**Pull request by kaydentsoftwire**
## What's changed
Added secondary sorting to sort methods in this order:
packing date > packing slot > deliveries then collections > collection centre name > postcode

(if one of these is the primary sort then this is moved to the top).

For each column:

- **Family category** = family category > packing date > packing slot > delivery/collection > collection centre name > postcode
- **Full name** = full name > packing date > packing slot > delivery/collection > collection centre name > postcode
- **Postcode** = postcode > packing date > packing slot > delivery/collection > collection centre name 
- **Phone number** = phone number > packing date > packing slot > delivery / collection > collection centre name > postcode
- **Voucher number** = voucher number > packing date > packing slot > delivery / collection > collection centre name > postcode 
- **Packing slot** = packing slot > packing date > delivery / collection > collection centre name > postcode 
- **Method** = method >  collection centre name > packing date > packing slot > postcode
- **Last status** = last status > packing date > packing slot > delivery / collection > collection centre name > postcode
- **Created at** = created at > packing date > packing slot > delivery / collection > collection centre name > postcode

The final sort for all columns is parcel id to ensure consistency 

## Screenshots / Videos
| Before     | After      |
|------------|------------|
| image | image|
| image |image|
| image |  image
|image | image |
| image | image 
| image | image |
| image | image |
| image | image

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA
- [x] I have self-reviewed this PR
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test:e2e`

### CODE REVIEW

File: src/app/parcels/parcelsTable/sortableColumns.ts [hunk 1]
```diff
@@ -7,47 +7,84 @@ export const defaultParcelsSort: ParcelsSortMethod = (query, sortDirection) =>
         .order("packing_slot_order")
         .order("is_delivery", { ascending: false })
         .order("client_is_active", { ascending: false })
+        .order("collection_centre_name")
         .order("client_address_postcode")
-        .order("collection_centre_name");
+        .order("parcel_id");
```
Comment (kaydentsoftwire) [inline src/app/parcels/parcelsTable/sortableColumns.ts:4]: Swapped these around as we want collection centre name to sort before postcodes

**Review by kaydentsoftwire — COMMENTED** [review]
Applied secondary sorting to all columns of the parcels table to match the default sorting, and added the final order of parcel id to ensure consistency across sorts

**Comment by AmeliaIsCoding** [comment]
Hi @kaydentsoftwire - just checking that you've merged with dev recently, as in my recent ticket I included sorting by "client_is_active" before sorting by any client information fields - this is required because if a client is inactive but has personal information (i.e., a seeded inactive client), we could expose the sensitive information by sorting. This should be included in these sorts now please :)

client information fields are anything in the backend clients table (eg postcode)

File: src/app/parcels/parcelsTable/ParcelsPage.tsx [hunk 2]
```diff
@@ -323,6 +323,7 @@ const ParcelsPage: React.FC<{}> = () => {
                                 sortableColumns: parcelsSortableColumns,
                                 setSortState: setSortState,
                             }}
+                            defaultSortField={defaultParcelsSortKey}
```
Comment (stonelink) [inline src/app/parcels/parcelsTable/ParcelsPage.tsx:326]: This `defaultSortField` and underlying Table.tsx changes are to resolve the outstanding request for visible default sorting on the Parcels page as per this comment: https://github.com/NorwoodAndBrixtonFoodbank/nbf-website/issues/152#issuecomment-2154830486

**Comment by stonelink** [comment]
The `defaultParcelsSortConfig` and changes are to fix the dividing lines behaviour - see this comment: https://github.com/NorwoodAndBrixtonFoodbank/nbf-website/issues/152#issuecomment-2154830486

**Review by stonelink — APPROVED** [review]

**Pull request merged**